### ISSUE TITLE
[flake8-simplify] SIM910/SIM911 false negative if variable is named dict

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#18777]
### Summary

`SIM911` has a false negative if the dict is named `dict` [playground](https://play.ruff.rs/856a5d41-547e-4537-8958-891966c3d97b)
```
PS ~\Desktop\New_folder>Get-Content issue.py
```
```py
dict = {}
for country, stars in zip(dict.keys(), dict.values()):...
```
```
PS ~\Desktop\New_folder>uvx ruff check issue.py --isolated --select SIM
```
```
All checks passed!
```

This also applies to `SIM910` [playground](https://play.ruff.rs/b76b7679-7f6f-4430-b694-f6629efa4efe)

### Version

ruff 0.12.0 (87f0feb 2025-06-17) + playground

**Comment by LaBatata101** [comment]
I'm working on this one
